include/deal.II/grid/grid_tools.h

```diff
@@ -1032,6 +1032,35 @@ namespace GridTools
   ( const MeshType                                                                    &mesh,
     const std::function<bool (const typename MeshType::active_cell_iterator &)> &predicate );
 
+  /**
+   * Compute a collection of bounding boxes so that all cells for which the given predicate is true, are
+   * completely enclosed in at least one of the bounding boxes.
+   *
+   * The algorithm loops over all cells at given level @p refinement_level and active cells at coarser refinement
+   * levels, creating a Bounding Box for each of them. Then it merges the boxes which can be expressed as a single
+   * one. In a last optional step all smaller boxes are merged with neighbors.
```

Comment:
I don't understand the last sentence, can you reformulate or expand the description? What are "smaller boxes"? Do you imply that a "merge" is a combination without error? You also seem to describe the algorithm to get the result and not the result in an abstract way. It is typically more helpful for a user to get a description of the output independent on how you achieve it. Maybe something like "a cover of the cells with the minimum number of non-overlapping bounding boxes"?

Reply:
I modified a lot the description, as you shall see shortly: I hope it's better...thank you for the comments!